contrib/format-httpd/README.md

```diff
@@ -41,24 +45,67 @@ FROM dfs.test.`logfile.httpd` AS mylogs
-field called `request_firstline_uri_query_username`.  
+In the event that you have a map field that you would like broken into columns rather than getting the nested fields, you can set the `flattenWildcards` option to `true` and
+Drill will create columns for these fields.  For example if you have a URI Query option called `username`.  If you selected the `flattedWildcards` option, Drill will create a
+field called `request_firstline_uri_query_username`.
 
-** Note that underscores in the field name are replaced with double underscores ** 
- 
- ## Useful Functions
+** Note that underscores in the field name are replaced with double underscores **
+
+## Useful Functions
  If you are using Drill to analyze web access logs, there are a few other useful functions which you should know about:
- 
+
  * `parse_url(<url>)`: This function accepts a URL as an argument and returns a map of the URL's protocol, authority, host, and path.
  * `parse_query(<query_string>)`: This function accepts a query string and returns a key/value pairing of the variables submitted in the request.
  * `parse_user_agent(<user agent>)`, `parse_user_agent( <useragent field>, <desired field> )`: The function parse_user_agent() takes a user agent string as an argument and
-  returns a map of the available fields. Note that not every field will be present in every user agent string. 
+  returns a map of the available fields. Note that not every field will be present in every user agent string.
   [Complete Docs Here](https://github.com/apache/drill/tree/master/contrib/udfs#user-agent-functions)
- 
+
+## LogParser type remapping
+**Advanced feature**
+The underlying [logparser](https://github.com/nielsbasjes/logparser) supports something called type remapping.
+Essentially it means that an extracted value which would normally be treated as an unparsable STRING can now be 'cast' to something
+that can be further cut into relevant pieces.
+
+The parameter string is a `;` separated list of mappings.
+Each mapping is a `:` separated list of
+- the name of the underlying logparser field (which is different from th Drill column name),
+- the underlying `type` which is used to determine which additional Dissectors can be applied.
+- optionally the `cast` (one of `STRING`, `LONG`, `DOUBLE`) which may impact the type of the Drill column
+
+Examples:
+- If you have a query parameter in the URL called `ua` which is really the UserAgent string and you would like to parse this you can add
+`request.firstline.uri.query.ua:HTTP.USERAGENT`
+- If you have a query parameter in the URL called `timestamp` which is really the numerical timestamp (epoch milliseconds).
+The additional "LONG" will cause the returned value be a long which tells Drill the `TIME.EPOCH` is to be interpreted as a `TIMESTAMP` column.
+`request.firstline.uri.query.timestamp:TIME.EPOCH:LONG`
+
```

Comment:
This is slick! 🥇

Reply:
Yes, I was hoping you would like this feature to be available in Drill.
Note that the name of the parameter does not matter. If it is called `foo` you can do `request.firstline.uri.query.foo:TIME.EPOCH:LONG` and it will work aswell.

Having this is especially useful if you are parsing web analytics `beacon` logs (where the real data is all in query parameters).
If you have a query parameter that is really the URL of your site then simply remap it to `HTTP.URI` and go from there.
`request.firstline.uri.query.g:HTTP.URI`

Another nice example of this is what I do here with the MOD_UNIQUE_ID field ... for which a built in Dissector is available:

https://github.com/apache/drill/blob/29edfec469c7de4cdd621fd4914dfb143d9a2784/contrib/format-httpd/src/test/java/org/apache/drill/exec/store/httpd/TestHTTPDLogReaderUserAgent.java#L194

Reply:
Have a look at this file as an example of what I mean with a `beacon` log:
https://github.com/nielsbasjes/logparser/blob/master/httpdlog/httpdlog-pigloader/src/test/resources/omniture-access.log
Here the `g` query parameter is the real URL ... which in itself has query parameters of its own that I'm interested in for my analysis.

This is the kind of logfile that actually triggered me to write this library all those years ago.